providers/stores/mongo.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const MongoClient = require('mongodb').MongoClient
+const promiseRetry = require('promise-retry')
+const EntityCoordinates = require('../../lib/entityCoordinates')
+
+class MongoStore {
+  constructor(options) {
+    this.options = options
+  }
+
+  initialize() {
+    return promiseRetry(async retry => {
+      try {
+        this.client = await MongoClient.connect(
+          this.options.connectionString,
+          { useNewUrlParser: true }
+        )
+        this.db = this.client.db(this.options.dbName)
+        this.collection = this.db.collection('definitions')
+      } catch (error) {
+        retry(error)
+      }
+    })
+  }
+
+  /**
+   * List all of the matching components for the given coordinates.
+   * Accepts partial coordinates.
+   *
+   * @param {EntityCoordinates} coordinates
+   * @returns A list of matching coordinates i.e. [ 'npm/npmjs/-/JSONStream/1.3.3' ]
+   */
+  async list(coordinates) {
+    const list = await this.collection.find({ id: this._getId(coordinates) }, 'id')
```

Comment:
this needs to be update to list by prefix, something like:

```js
await this.collection.find({ id: new RegExp('^' + this._getId(coordinates) }, 'id')
```

Reply:
Regex does work for mongo filters. 

Good point about needing to escape I hadn't thought of that.

For the redos protection, maybe we can do something simple like limit the length of the input to 100 chars

Reply:
Not sure. This library looks legit https://github.com/substack/safe-regex/blob/master/readme.markdown